medschool/src/types.rs

```diff
@@ -0,0 +1,180 @@
+    #[tracing::instrument(level = "trace", ret)]
+    pub fn new(field: syn::Field) -> Option<Self> {
+        let type_ident = match field.ty {
+            Type::Path(type_path) => {
+                // `get_ident` returns `Some` if the `path` doesn't contain generics.
+                type_path
+                    .path
+                    .get_ident()
+                    .cloned()
+                    .or_else(|| get_ident_from_field_skipping_generics(type_path))
+            }
+            _ => None,
+        }?;
+
+        let mut docs = docs_from_attributes(field.attrs);
+
+        // TODO: this part can be encapsulated into a function
+        for doc in docs.iter_mut() {
+            // removes docs that we don't want in `configuration.md`
+            if doc.contains(r"<!--${internal}-->") {
+                return None;
+            }
+
+            // `trim` is too aggressive, we just want to remove 1 whitespace
+            if doc.starts_with(' ') {
+                doc.remove(0);
+            }
+        }
+
+        docs.push("\n".to_string());
+
+        Some(Self {
+            ident: field.ident?.to_string(),
+            ty: type_ident.to_string(),
+            docs,
+        })
+    }
+}
+
+impl PartialType {
+    /// Turns the `root` [`PartialType`] documentation into one big `String`.
+    pub fn produce_docs(self) -> String {
+        self.docs
+            .into_iter()
+            .chain(
+                self.fields
+                    .into_iter()
+                    .flat_map(|field| field.docs.into_iter()),
+            )
+            .collect()
+    }
+}
+
+impl PartialEq for PartialField {
+    fn eq(&self, other: &Self) -> bool {
+        self.ident == other.ident
+    }
+}
+
+impl Eq for PartialField {}
+
+impl PartialOrd for PartialField {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for PartialField {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.ident.cmp(&other.ident)
+    }
+}
+
+impl From<PartialField> for PartialType {
+    fn from(value: PartialField) -> Self {
+        Self {
+            ident: value.ident,
+            docs: value.docs,
+            fields: Default::default(),
+        }
+    }
+}
+
+impl PartialEq for PartialType {
+    fn eq(&self, other: &Self) -> bool {
+        self.ident.eq(&other.ident)
+    }
+}
+
+impl Eq for PartialType {}
+
+impl Hash for PartialType {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.ident.hash(state)
+    }
+}
+
+impl Borrow<String> for PartialType {
+    fn borrow(&self) -> &String {
+        &self.ident
+    }
+}
+
+impl Display for PartialType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        // Prevents crashing the tool if the type is not properly documented.
+        let docs = self.docs.last().cloned().unwrap_or_default();
+        f.write_str(&docs)?;
+
+        self.fields.iter().fold(f, |accum, s| {
```

Comment:
```suggestion
        self.fields.iter().fold(f, |final_docs, type_| {
```

Reply:
Or something along those lines, let's avoid single chars and names that are too generic.